### ISSUE TITLE
Update GitLab CI to 11.9.1

### ISSUE THREAD

**Issue by tnir** [tnir/mysqlclient#59]
cf. #48 (#47)

**Comment by tnir** [comment]
```diff
$ diff -u <(curl -sSL https://gitlab.com/gitlab-org/gitlab-ce/raw/v11.7.5/lib/gitlab/ci/templates/Auto-DevOps.gitlab-ci.yml) <(curl -sSL https://gitlab.com/gitlab-org/gitlab-ce/raw/v11.9.1/lib/gitlab/ci/templates/Auto-DevOps.gitlab-ci.yml)
--- /dev/fd/63  2019-04-01 10:14:45.000000000 +0900
+++ /dev/fd/62  2019-04-01 10:14:45.000000000 +0900
@@ -21,8 +21,8 @@
 #
 # In order to deploy, you must have a Kubernetes cluster configured either
 # via a project integration, or via group/project variables.
-# AUTO_DEVOPS_DOMAIN must also be set as a variable at the group or project
-# level, or manually added below.
+# KUBE_INGRESS_BASE_DOMAIN must also be set on the cluster settings,
+# as a variable at the group or project level, or manually added below.
 #
 # Continuous deployment to production is enabled by default.
 # If you want to deploy to staging first, set STAGING_ENABLED environment variable.
@@ -41,19 +41,22 @@
 image: alpine:latest
 
 variables:
-  # AUTO_DEVOPS_DOMAIN is the application deployment domain and should be set as a variable at the group or project level.
-  # AUTO_DEVOPS_DOMAIN: domain.example.com
+  # KUBE_INGRESS_BASE_DOMAIN is the application deployment domain and should be set as a variable at the group or project level.
+  # KUBE_INGRESS_BASE_DOMAIN: domain.example.com
 
   POSTGRES_USER: user
   POSTGRES_PASSWORD: testing-password
   POSTGRES_ENABLED: "true"
   POSTGRES_DB: $CI_ENVIRONMENT_SLUG
+  POSTGRES_VERSION: 9.6.2
 
-  KUBERNETES_VERSION: 1.10.9
-  HELM_VERSION: 2.11.0
+  KUBERNETES_VERSION: 1.11.7
+  HELM_VERSION: 2.12.3
 
   DOCKER_DRIVER: overlay2
 
+  ROLLOUT_RESOURCE_TYPE: deployment
+
 stages:
   - build
   - test
@@ -71,14 +74,14 @@
 
 build:
   stage: build
-  image: docker:stable-git
+  image: "registry.gitlab.com/gitlab-org/cluster-integration/auto-build-image/master:stable"
   services:
-  - docker:stable-dind
+    - docker:stable-dind
   script:
-    - setup_docker
-    - build
+    - /build/build.sh
   only:
     - branches
+    - tags
 
 test:
   services:
@@ -93,6 +96,7 @@
     - /bin/herokuish buildpack test
   only:
     - branches
+    - tags
   except:
     variables:
       - $TEST_DISABLED
@@ -110,13 +114,14 @@
     paths: [gl-code-quality-report.json]
   only:
     - branches
+    - tags
   except:
     variables:
       - $CODE_QUALITY_DISABLED
 
 license_management:
   stage: test
-  image: 
+  image:
     name: "registry.gitlab.com/gitlab-org/security-products/license-management:$CI_SERVER_VERSION_MAJOR-$CI_SERVER_VERSION_MINOR-stable"
     entrypoint: [""]
   allow_failure: true
@@ -127,6 +132,7 @@
   only:
     refs:
       - branches
+      - tags
     variables:
       - $GITLAB_FEATURES =~ /\blicense_management\b/
   except:
@@ -149,6 +155,7 @@
   only:
     refs:
       - branches
+      - tags
     kubernetes: active
   except:
     variables:
@@ -169,6 +176,7 @@
   only:
     refs:
       - branches
+      - tags
     variables:
       - $GITLAB_FEATURES =~ /\bsast\b/
   except:
@@ -190,6 +198,7 @@
   only:
     refs:
       - branches
+      - tags
     variables:
       - $GITLAB_FEATURES =~ /\bdependency_scanning\b/
   except:
@@ -210,6 +219,7 @@
   only:
     refs:
       - branches
+      - tags
     variables:
       - $GITLAB_FEATURES =~ /\bcontainer_scanning\b/
   except:
@@ -229,6 +239,7 @@
   only:
     refs:
       - branches
+      - tags
     kubernetes: active
     variables:
       - $GITLAB_FEATURES =~ /\bdast\b/
@@ -251,13 +262,14 @@
     - persist_environment_url
   environment:
     name: review/$CI_COMMIT_REF_NAME
-    url: http://$CI_PROJECT_PATH_SLUG-$CI_ENVIRONMENT_SLUG.$AUTO_DEVOPS_DOMAIN
+    url: http://$CI_PROJECT_PATH_SLUG-$CI_ENVIRONMENT_SLUG.$KUBE_INGRESS_BASE_DOMAIN
     on_stop: stop_review
   artifacts:
     paths: [environment_url.txt]
   only:
     refs:
       - branches
+      - tags
     kubernetes: active
   except:
     refs:
@@ -281,6 +293,7 @@
   only:
     refs:
       - branches
+      - tags
     kubernetes: active
   except:
     refs:
@@ -306,7 +319,7 @@
     - deploy
   environment:
     name: staging
-    url: http://$CI_PROJECT_PATH_SLUG-staging.$AUTO_DEVOPS_DOMAIN
+    url: http://$CI_PROJECT_PATH_SLUG-staging.$KUBE_INGRESS_BASE_DOMAIN
   only:
     refs:
       - master
@@ -330,7 +343,7 @@
     - deploy canary
   environment:
     name: production
-    url: http://$CI_PROJECT_PATH_SLUG.$AUTO_DEVOPS_DOMAIN
+    url: http://$CI_PROJECT_PATH_SLUG.$KUBE_INGRESS_BASE_DOMAIN
   when: manual
   only:
     refs:
@@ -354,7 +367,7 @@
     - persist_environment_url
   environment:
     name: production
-    url: http://$CI_PROJECT_PATH_SLUG.$AUTO_DEVOPS_DOMAIN
+    url: http://$CI_PROJECT_PATH_SLUG.$KUBE_INGRESS_BASE_DOMAIN
   artifacts:
     paths: [environment_url.txt]
 
@@ -403,7 +416,7 @@
     - persist_environment_url
   environment:
     name: production
-    url: http://$CI_PROJECT_PATH_SLUG.$AUTO_DEVOPS_DOMAIN
+    url: http://$CI_PROJECT_PATH_SLUG.$KUBE_INGRESS_BASE_DOMAIN
   artifacts:
     paths: [environment_url.txt]
 
@@ -486,9 +499,13 @@
   [[ "$TRACE" ]] && set -x
   auto_database_url=postgres://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${CI_ENVIRONMENT_SLUG}-postgres:5432/${POSTGRES_DB}
   export DATABASE_URL=${DATABASE_URL-$auto_database_url}
-  export CI_APPLICATION_REPOSITORY=$CI_REGISTRY_IMAGE/$CI_COMMIT_REF_SLUG
-  export CI_APPLICATION_TAG=$CI_COMMIT_SHA
-  export CI_CONTAINER_NAME=ci_job_build_${CI_JOB_ID}
+  if [[ -z "$CI_COMMIT_TAG" ]]; then
+    export CI_APPLICATION_REPOSITORY=$CI_REGISTRY_IMAGE/$CI_COMMIT_REF_SLUG
+    export CI_APPLICATION_TAG=$CI_COMMIT_SHA
+  else
+    export CI_APPLICATION_REPOSITORY=$CI_REGISTRY_IMAGE
+    export CI_APPLICATION_TAG=$CI_COMMIT_TAG
+  fi
   export TILLER_NAMESPACE=$KUBE_NAMESPACE
   # Extract "MAJOR.MINOR" from CI_SERVER_VERSION and generate "MAJOR-MINOR-stable" for Security Products
   export SP_VERSION=$(echo "$CI_SERVER_VERSION" | sed 's/^\([0-9]*\)\.\([0-9]*\).*/\1-\2-stable/')
@@ -611,16 +628,16 @@
     track="${1-stable}"
     export APPLICATION_SECRET_NAME=$(application_secret_name "$track")
 
-    bash -c '
-      function k8s_prefixed_variables() {
-        env | sed -n "s/^K8S_SECRET_\(.*\)$/\1/p"
-      }
-
-      kubectl create secret \
-        -n "$KUBE_NAMESPACE" generic "$APPLICATION_SECRET_NAME" \
-        --from-env-file <(k8s_prefixed_variables) -o yaml --dry-run |
-        kubectl replace -n "$KUBE_NAMESPACE" --force -f -
-    '
+    env | sed -n "s/^K8S_SECRET_\(.*\)$/\1/p" > k8s_prefixed_variables
+
+    kubectl create secret \
+      -n "$KUBE_NAMESPACE" generic "$APPLICATION_SECRET_NAME" \
+      --from-env-file k8s_prefixed_variables -o yaml --dry-run |
+      kubectl replace -n "$KUBE_NAMESPACE" --force -f -
+
+    export APPLICATION_SECRET_CHECKSUM=$(cat k8s_prefixed_variables | sha256sum | cut -d ' ' -f 1)
+
+    rm k8s_prefixed_variables
   }
 
   function deploy_name() {
@@ -667,6 +684,14 @@
 
     create_application_secret "$track"
 
+    env_slug=$(echo ${CI_ENVIRONMENT_SLUG//-/_} | tr -s '[:lower:]' '[:upper:]')
+    eval env_ADDITIONAL_HOSTS=\$${env_slug}_ADDITIONAL_HOSTS
+    if [ -n "$env_ADDITIONAL_HOSTS" ]; then
+      additional_hosts="{$env_ADDITIONAL_HOSTS}"
+    elif [ -n "$ADDITIONAL_HOSTS" ]; then
+      additional_hosts="{$ADDITIONAL_HOSTS}"
+    fi
+
     if [[ -n "$DB_INITIALIZE" && -z "$(helm ls -q "^$name$")" ]]; then
       echo "Deploying first release with database initialization..."
       helm upgrade --install \
@@ -680,13 +705,17 @@
         --set application.track="$track" \
         --set application.database_url="$DATABASE_URL" \
         --set application.secretName="$APPLICATION_SECRET_NAME" \
+        --set application.secretChecksum="$APPLICATION_SECRET_CHECKSUM" \
+        --set service.commonName="le.$KUBE_INGRESS_BASE_DOMAIN" \
         --set service.url="$CI_ENVIRONMENT_URL" \
+        --set service.additionalHosts="$additional_hosts" \
         --set replicaCount="$replicas" \
         --set postgresql.enabled="$postgres_enabled" \
         --set postgresql.nameOverride="postgres" \
         --set postgresql.postgresUser="$POSTGRES_USER" \
         --set postgresql.postgresPassword="$POSTGRES_PASSWORD" \
         --set postgresql.postgresDatabase="$POSTGRES_DB" \
+        --set postgresql.imageTag="$POSTGRES_VERSION" \
         --set application.initializeCommand="$DB_INITIALIZE" \
         --namespace="$KUBE_NAMESPACE" \
         "$name" \
@@ -713,7 +742,10 @@
         --set application.track="$track" \
         --set application.database_url="$DATABASE_URL" \
         --set application.secretName="$APPLICATION_SECRET_NAME" \
+        --set application.secretChecksum="$APPLICATION_SECRET_CHECKSUM" \
+        --set service.commonName="le.$KUBE_INGRESS_BASE_DOMAIN" \
         --set service.url="$CI_ENVIRONMENT_URL" \
+        --set service.additionalHosts="$additional_hosts" \
         --set replicaCount="$replicas" \
         --set postgresql.enabled="$postgres_enabled" \
         --set postgresql.nameOverride="postgres" \
@@ -726,7 +758,7 @@
         chart/
     fi
 
-    kubectl rollout status -n "$KUBE_NAMESPACE" -w "deployment/$name"
+    kubectl rollout status -n "$KUBE_NAMESPACE" -w "$ROLLOUT_RESOURCE_TYPE/$name"
   }
 
   function scale() {
@@ -793,7 +825,7 @@
     fi
 
     helm init --client-only
-    helm repo add gitlab https://charts.gitlab.io
+    helm repo add gitlab ${AUTO_DEVOPS_CHART_REPOSITORY:-https://charts.gitlab.io}
     if [[ ! -d "$auto_chart" ]]; then
       helm fetch ${auto_chart} --untar
     fi
@@ -809,59 +841,28 @@
     kubectl describe namespace "$KUBE_NAMESPACE" || kubectl create namespace "$KUBE_NAMESPACE"
   }
 
-  function check_kube_domain() {
-    if [ -z ${AUTO_DEVOPS_DOMAIN+x} ]; then
-      echo "In order to deploy or use Review Apps, AUTO_DEVOPS_DOMAIN variable must be set"
-      echo "You can do it in Auto DevOps project settings or defining a variable at group or project level"
-      echo "You can also manually add it in .gitlab-ci.yml"
-      false
-    else
-      true
+
+  # Function to ensure backwards compatibility with AUTO_DEVOPS_DOMAIN
+  function ensure_kube_ingress_base_domain() {
+    if [ -z ${KUBE_INGRESS_BASE_DOMAIN+x} ] && [ -n "$AUTO_DEVOPS_DOMAIN" ] ; then
+      export KUBE_INGRESS_BASE_DOMAIN=$AUTO_DEVOPS_DOMAIN
     fi
   }
 
-  function build() {
-    registry_login
+  function check_kube_domain() {
+    ensure_kube_ingress_base_domain
 
-    if [[ -f Dockerfile ]]; then
-      echo "Building Dockerfile-based application..."
-      docker build \
-        --build-arg HTTP_PROXY="$HTTP_PROXY" \
-        --build-arg http_proxy="$http_proxy" \
-        --build-arg HTTPS_PROXY="$HTTPS_PROXY" \
-        --build-arg https_proxy="$https_proxy" \
-        --build-arg FTP_PROXY="$FTP_PROXY" \
-        --build-arg ftp_proxy="$ftp_proxy" \
-        --build-arg NO_PROXY="$NO_PROXY" \
-        --build-arg no_proxy="$no_proxy" \
-        -t "$CI_APPLICATION_REPOSITORY:$CI_APPLICATION_TAG" .
+    if [ -z ${KUBE_INGRESS_BASE_DOMAIN+x} ]; then
+      echo "In order to deploy or use Review Apps,"
+      echo "AUTO_DEVOPS_DOMAIN or KUBE_INGRESS_BASE_DOMAIN variables must be set"
+      echo "From 11.8, you can set KUBE_INGRESS_BASE_DOMAIN in cluster settings"
+      echo "or by defining a variable at group or project level."
+      echo "You can also manually add it in .gitlab-ci.yml"
+      echo "AUTO_DEVOPS_DOMAIN support will be dropped on 12.0"
+      false
     else
-      echo "Building Heroku-based application using gliderlabs/herokuish docker image..."
-      docker run -i \
-        -e BUILDPACK_URL \
-        -e HTTP_PROXY \
-        -e http_proxy \
-        -e HTTPS_PROXY \
-        -e https_proxy \
-        -e FTP_PROXY \
-        -e ftp_proxy \
-        -e NO_PROXY \
-        -e no_proxy \
-        --name="$CI_CONTAINER_NAME" -v "$(pwd):/tmp/app:ro" gliderlabs/herokuish /bin/herokuish buildpack build
-      docker commit "$CI_CONTAINER_NAME" "$CI_APPLICATION_REPOSITORY:$CI_APPLICATION_TAG"
-      docker rm "$CI_CONTAINER_NAME" >/dev/null
-      echo ""
-
-      echo "Configuring $CI_APPLICATION_REPOSITORY:$CI_APPLICATION_TAG docker image..."
-      docker create --expose 5000 --env PORT=5000 --name="$CI_CONTAINER_NAME" "$CI_APPLICATION_REPOSITORY:$CI_APPLICATION_TAG" /bin/herokuish procfile start web
-      docker commit "$CI_CONTAINER_NAME" "$CI_APPLICATION_REPOSITORY:$CI_APPLICATION_TAG"
-      docker rm "$CI_CONTAINER_NAME" >/dev/null
-      echo ""
+      true
     fi
-
-    echo "Pushing to GitLab Container Registry..."
-    docker push "$CI_APPLICATION_REPOSITORY:$CI_APPLICATION_TAG"
-    echo ""
   }
 
   function initialize_tiller() {
```

**Comment by tnir** [comment]
Closed in favor of #65
